### PR TITLE
DOC: Add bcpandas to Ecosystem in docs

### PR DESCRIPTION
**Pull request by yehoshuadimarsky**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Hi, I wrote this utility which I use in production almost daily at my job, and it's been super helpful for me personally to fill a gap - slow writes from Pandas to MS SQL. Figured I'd suggest it in the pandas ecosystem if it could be useful to others.

Wasn't sure where to put it, and didn't want to start a new section, so I put it in `Out-of-core`. It appeared to be sorted alphabetically, so I put it in that order.

### CODE REVIEW

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -325,6 +325,17 @@ dimensional arrays, rather than the tabular data for which pandas excels.
 Out-of-core
 -------------
 
+`BCPandas <https://github.com/yehoshuadimarsky/bcpandas>`__
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst:328]: I would make a new section about IO, as this doesn't really fit for "out-of-core" ? 
You can eg put it just above "out of core"

File: doc/source/ecosystem.rst [hunk 2]
```diff
@@ -325,6 +325,17 @@ dimensional arrays, rather than the tabular data for which pandas excels.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+BCPandas provides high performance writes from pandas to Microsoft SQL Server,
+far exceeding the performance of the native ``df.to_sql`` method. Internally, it uses
+Microsoft's BCP utility, but the complexity is fully abstracted away from the end user.
+Rigorously tested, it is a complete replacement for ``df.to_sql``.
+
+(Testing has shown that the same peformance speedup is not achieved when reading data,
+only when writing data, so it is not a full I/O replacement, just the "I" part).
```
Comment (jorisvandenbossche) [inline doc/source/ecosystem.rst:337]: I would leave this out, and just clearly note that it only provides a `to_sql` alternative

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for adding this!

**Comment by yehoshuadimarsky** [comment]
Just made those changes thanks

**Comment by jreback** [comment]
@yehoshuadimarsky can you update to @jorisvandenbossche comments, otherwise lgtm.

File: doc/source/ecosystem.rst (hunk 2 above)
Comment (yehoshuadimarsky) [reply doc/source/ecosystem.rst]: ok did

**Comment by yehoshuadimarsky** [comment]
@jreback ok just did, thanks

**Comment by jreback** [comment]
thanks @yehoshuadimarsky

**Pull request merged**